components/slack/actions/add_star/add_star.js

```diff
@@ -0,0 +1,42 @@
+const slack = require("../../slack.app.js");
+const { WebClient } = require("@slack/web-api");
+
+module.exports = {
+  key: "slack-add-star",
+  name: "Add Star",
+  description: "Add a star to an item on behalf of the authenticated user",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    slack,
+    conversation: {
+      propDefinition: [
+        slack,
+        "conversation",
+      ],
+      optional: true,
+    },
+    timestamp: {
+      propDefinition: [
+        slack,
+        "timestamp",
+      ],
+      optional: true,
+    },
+    file: {
+      propDefinition: [
+        slack,
+        "file",
+      ],
+      optional: true,
+    },
+  },
+  async run() {
+    const web = new WebClient(this.slack.$auth.oauth_access_token);
```

Comment:
Anywhere you use the Slack client, I'd recommend using [the `sdk()` method from the app file](https://github.com/PipedreamHQ/pipedream/blob/b7fac86f89f9076c80686886f0875cece62c0290/components/slack/slack.app.js#L322). This helps in two ways:

- Keeps the code DRY, and you don't have to `require` the Slack client within each component
- It's possible the Slack integration will change in the future and the OAuth access token will be in a different property than `this.slack.$auth.oauth_access_token`, so we define it once in `sdk()`.